react/ListPriceRange.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { useContext } from 'react'
+import { defineMessages } from 'react-intl'
+import { ProductContext } from 'vtex.product-context'
+import { FormattedCurrency } from 'vtex.format-currency'
+import { useCssHandles } from 'vtex.css-handles'
+import { IOMessageWithMarkers } from 'vtex.native-types'
+
+import { StorefrontFC, PriceRangeProps } from './types'
+
+const CSS_HANDLES = [
+  'listPriceRange',
+  'listPriceRangeMinValue',
+  'listPriceRangeMaxValue',
+] as const
+
+const ListPriceRange: StorefrontFC<PriceRangeProps> = props => {
+  const { message, noRangeMessage, markers } = props
+  const handles = useCssHandles(CSS_HANDLES)
+  const { product } = useContext(ProductContext)
+
+  const priceRange = product?.priceRange
+  if (!priceRange) {
+    return null
+  }
+
+  if (
+    priceRange.listPrice.lowPrice === priceRange.sellingPrice.lowPrice &&
+    priceRange.listPrice.highPrice === priceRange.sellingPrice.highPrice
+  ) {
+    return null
+  }
+
+  const minPrice = priceRange.listPrice.lowPrice
+  const maxPrice = priceRange.listPrice.highPrice
+  const hasRange = minPrice !== maxPrice
+
+  return (
+    <span className={handles.listPriceRange}>
+      <IOMessageWithMarkers
+        message={hasRange ? message : noRangeMessage}
+        markers={markers}
+        handleBase="listPriceRange"
+        values={{
+          minPriceValue: (
+            <span className={`${handles.listPriceRangeMinValue} strike`}>
+              <FormattedCurrency value={minPrice} />
+            </span>
+          ),
+          maxPriceValue: (
+            <span className={`${handles.listPriceRangeMaxValue} strike`}>
+              <FormattedCurrency value={maxPrice} />
+            </span>
+          ),
```

Comment:
What do you think in the case of not existing a range, only make the variable `listPriceValue` available?